central/image/datastore/store/v2/postgres/store.go

```diff
@@ -557,6 +560,17 @@ func (s *storeImpl) populateImage(ctx context.Context, tx *postgres.Tx, image *s
 			cveParts = append(cveParts, cvePart)
 		}
 
+		// TODO(ROX-27399):  Adding the index of where the vuln appeared in the component
+		// is not likely sustainable.  We cannot easily guarantee the order is the same when
+		// we pull the data out.  This sort is temporary to keep moving, and will be
+		// removed when the ID of the CVE is adjusted to no longer use the index of where
+		// the CVE occurs in the component list.
+		sort.SliceStable(cveParts, func(i, j int) bool {
+			cveICompIndex := getCVEComponentIndex(cveParts[i].CVEV2.GetId())
+			cveJCompIndex := getCVEComponentIndex(cveParts[j].CVEV2.GetId())
+			return cveICompIndex < cveJCompIndex
+		})
```

Comment:
From `sort.SliceStable`'s docs:

> Note: in many situations, the newer [slices.SortStableFunc](https://pkg.go.dev/slices#SortStableFunc) function is more ergonomic and runs faster.

```suggestion
		slices.SortStableFunc(cveParts, func(cvePartA, cvePartB common.CVEParts) bool {
			cveICompIndex := getCVEComponentIndex(cvePartA.CVEV2.GetId())
			cveJCompIndex := getCVEComponentIndex(cvePartB.CVEV2.GetId())
			return cveICompIndex < cveJCompIndex
		})
```